cmake/detray-functions.cmake

```diff
@@ -19,13 +19,19 @@ function( detray_add_library fullname basename )
    # Set up how clients should find its headers.
    target_include_directories( ${fullname} INTERFACE
       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-      $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}> )
+      $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}> )
```

Comment:
Don't add this here. We don't need to add the extra directory for all libraries, but only for `detray::core`. So let's just add a `target_include_directories(...)` call specifically for `detray::core`.